include/dali/core/convert.h

```diff
@@ -77,14 +77,37 @@ struct ret_type {  // a placeholder for return type
 
 template <typename T, typename U>
 DALI_HOST_DEV constexpr std::enable_if_t<
-    needs_clamp<U, T>::value && std::is_signed<U>::value,
+    needs_clamp<U, T>::value && std::is_signed<U>::value && std::is_signed<T>::value,
     T>
 clamp(U value, ret_type<T>) {
   return value <= min_value<T>() ? min_value<T>() :
          value >= max_value<T>() ? max_value<T>() :
          static_cast<T>(value);
 }
 
+template <typename T, typename U>
```

Comment:
This one is exacly the same as the one above. Was it needed then?

Reply:
There was an issue for some T unsigned case that need special handling, as compiler was complaining for signed/unsigned comparisons, you also cannot make and fp type unsigned.